src/main/java/duchess/storage/Storage.java

```diff
@@ -0,0 +1,102 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * The {@code Storage} class helps to save and load @{code ArrayList}s of
+ * {@code Task}s from a given File Path.
+ */
+public class Storage {
+    private String filePath;
+    private Gson gson;
+
+    /**
+     * Initialises a {@code Storage} instance that works with the given
+     * {@code filePath}.
+     *
+     * @param filePath The file path to save to and if possible, load from.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.gson = new Gson();
+    }
+
+    /**
+     * Saves a given list of tasks to the file path.
+     *
+     * @param tasks List of tasks to be saved.
+     * @throws DuchessException If it fails to save to the file path.
+     */
+    public void save(TaskList tasks) throws DuchessException {
+        try {
+            FileWriter fileWriter = new FileWriter(this.filePath);
+            Task[] taskArray = tasks.getTaskArray().toArray(new Task[tasks.size()]);
+            fileWriter.write(this.gson.toJson(taskArray, Task[].class));
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new DuchessException("Facing difficulties saving your tasks right now.");
+        }
+    }
+
+    /**
+     * Loads and returns a list of tasks from the file path.
+     *
+     * @return Loaded list of tasks from given file path.
+     * @throws DuchessException If it fails to load from the file path.
+     */
+    public ArrayList<Task> load() throws DuchessException {
+        try {
+            String fileContent = Files.readString(Path.of(this.filePath));
+            JsonArray array = JsonParser.parseString(fileContent).getAsJsonArray();
+            ArrayList<Task> result = new ArrayList<>();
+            for (int i = 0; i < array.size(); i++) {
+                JsonObject taskToCheck = (JsonObject) array.get(i);
+                if (taskToCheck.has("deadline")) {
+                    result.add(new Deadline(this.gson.fromJson(taskToCheck.get("description"), String.class),
+                            this.gson.fromJson(taskToCheck.get("deadline"), LocalDateTime.class),
+                            this.gson.fromJson(taskToCheck.get("isCompleted"), boolean.class)));
+                } else if (taskToCheck.has("timeFrame")) {
+                    result.add(new Event(this.gson.fromJson(taskToCheck.get("description"), String.class),
+                            this.gson.fromJson(taskToCheck.get("timeFrame"), String.class),
+                            this.gson.fromJson(taskToCheck.get("isCompleted"), boolean.class)));
+                } else {
+                    result.add(new ToDo(this.gson.fromJson(taskToCheck.get("description"), String.class),
+                            this.gson.fromJson(taskToCheck.get("isCompleted"), boolean.class)));
```

Comment:
Since `String` literals such as `isCompleted` are commonly used, you may want to store them as constants. 
Generally, we prefer avoid using `Strings` the middle of a block of code without explanation. You can read more here: https://help.semmle.com/wiki/display/JAVA/Magic+strings%3A+use+defined+constant

Reply:
Hi @raivatshah, thanks for the feedback and the awesome recommendation! I'll look into this and fix this up asap!

Reply:
Fixed in commit 9bed09b! Will not be resolving this conversation just in case it affects the grading bot.